instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/RocketMqConsumerTracer.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.rocketmq;
+
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER;
+import static io.opentelemetry.instrumentation.rocketmq.TextMapExtractAdapter.GETTER;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.List;
+import org.apache.rocketmq.client.consumer.listener.ConsumeConcurrentlyStatus;
+import org.apache.rocketmq.client.consumer.listener.ConsumeOrderlyStatus;
+import org.apache.rocketmq.common.message.MessageExt;
+
+public class RocketMqConsumerTracer extends BaseTracer {
+
+  private static final RocketMqConsumerTracer TRACER = new RocketMqConsumerTracer();
+
+  public static RocketMqConsumerTracer tracer() {
+    return TRACER;
+  }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.javaagent.rocketmq-client";
+  }
+
+  public Span startSpan(List<MessageExt> msgs) {
+    MessageExt msg = msgs.get(0);
```

Comment:
What if there are >1 messages? The rest won't be traced at all?

Reply:
When msgs. size( )>1, do we create one span or  several span? 
1.We create several span,but their "SemanticAttributes" vaules are the same. 
2. Create one span, whose span name is marked that this span processes multiple data, for example : test-topic process size(or batch) 10 
I think the second one is better. What's your opinion?

Reply:
Hmm, I think that messages may have different attributes, like different message id or queue offset.
Maybe we can do it the same way AWS SQS lambda handler instrumentation does in this scenario:

* create one root span for the whole message listener (without a parent);
* for each message create a child span and add a link to the producer span (extracted from the message headers).

This way each message will get its own span containing a link to propagated producer span, and the whole consumer processing will be wrapped in a single span. WDYT? @anuraaga

Reply:
Oh, and you can check out `TracingSqsEventHandler`, `TracingSqsMessageHandler` and `AwsLambdaMessageTracer` for an example of this.

Reply:
This situation is also described in the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#batch-receiving